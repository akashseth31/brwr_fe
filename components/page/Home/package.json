{
  "name": "HomePage",
  "version": "0.0.0",
  "private": true,
  "main": "./Home.page.js"
}
