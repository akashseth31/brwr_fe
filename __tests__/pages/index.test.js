import { render, screen } from '@testing-library/react';
import Home from '../../pages/index';
import '@testing-library/jest-dom';

describe('Home', () => {
  let component;

  beforeEach(() => {
    component = render(<Home />);
  });

  it('should render', () => {
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('renders a heading', () => {
    const heading = screen.getByRole('heading', {
      name: /home page/i,
    })

    expect(heading).toBeInTheDocument();
  });

  it('renders a link', () => {
    const link = screen.getByRole('link', {
      name: 'Go to project',
    });

    expect(link).toHaveAttribute('href', '/project');

    expect(link).toBeInTheDocument();
  });
});
