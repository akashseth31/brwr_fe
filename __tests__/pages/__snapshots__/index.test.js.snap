// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render 1`] = `
<DocumentFragment>
  <h1>
    Home Page
  </h1>
  <a
    href="/project"
  >
    Go to project
  </a>
</DocumentFragment>
`;
